### PR TITLE
Global settings, associations

### PR DESCRIPTION
**Pull request by Piwakk**
Create / edit / delete an association
The library and marketplace are not removed when their association is removed, which is weird

### CODE REVIEW

File: backend/associations/serializers/association.py [hunk 1]
```diff
@@ -1,43 +1,23 @@
...
-    pages = serializers.SerializerMethodField()
-    marketplace = MarketplaceShortSerializer(read_only=True)
-    library = LibraryShortSerializer(read_only=True)
     my_role = serializers.SerializerMethodField(read_only=True)
 
     class Meta:
         model = Association
-        read_only_fields = ("pages", "marketplace", "library", "my_role")
-        fields = read_only_fields + ("id", "name", "logo", "is_hidden", "rank")
+        read_only_fields = ("id", "my_role", "marketplace_enabled", "library_enabled")
```
Comment (abocquet) [inline backend/associations/serializers/association.py:14]: Maybe put these properties in a "enabled" field which contains marketplace and library, to add other functionalities in the futur (such  as medias for instance)

File: frontend/src/components/settings/associations/MutateAssociationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    initialValues={initialValues}
+    validationSchema={Yup.object({
+      name: Yup.string().required("Ce champ est requis."),
+      rank: Yup.number()
+        .integer("Le rang doit être un nombre entier.")
+        .required("Ce champ est requis."),
+    })}
+    onSubmit={(values, { resetForm, setSubmitting }) =>
+      onSubmit(values, { resetForm, setSubmitting })
+    }
```
Comment (abocquet) [inline frontend/src/components/settings/associations/MutateAssociationForm.tsx:30]:
```suggestion
    onSubmit={onSubmit}
```

File: frontend/src/components/settings/associations/SettingsAssociations.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        </Link>
+      </OverlayTrigger>
+    ),
+  },
+];
+
+export const SettingsAssociations = () => {
+  const { columns, sorting } = useColumns(columnsData);
+
+  return (
```
Comment (abocquet) [inline frontend/src/components/settings/associations/SettingsAssociations.tsx:59]: Maybe check  if the  user is an admin

Comment (Piwakk) [reply frontend/src/components/settings/associations/SettingsAssociations.tsx:59]: Done already in SettingsLayout

File: backend/tags/tests/test_hiding.py [hunk 4]
```diff
@@ -399,30 +399,6 @@ def test_no_media_in_association(self):
         self.assertNotIn("media", response.data, msg=msg)
         self.assertNotIn("medias", response.data, msg=msg)
 
-    def test_hiding_nested_pages_from_association(self):
```
Comment (abocquet) [inline backend/tags/tests/test_hiding.py:402]: Why do you remove this test ?

**Pull request merged**